test/ranges.jl

```diff
@@ -277,29 +277,49 @@ end
 @test all(((1:5) - [1:5;]) .== 0)
 
 # tricky floating-point ranges
+for (start, step, stop, len) in ((1, 1, 3, 3), (0, 1, 3, 4),
+                                 (3, -1, -1, 5), (1, -1, -3, 5),
+                                 (0, 1, 10, 11), (0, 7, 21, 4),
+                                 (0, 11, 33, 4), (1, 11, 34, 4),
+                                 (0, 13, 39, 4), (1, 13, 40, 4),
+                                 (11, 11, 33, 3), (3, 1, 11, 9),
+                                 (0, 10, 0, 1), (0, -1, 0, 1),
+                                 (0, 10, 55, 0), (0, -1, 5, 0),
+                                 (0, 1, 5, 0))
+    r = start/10:step/10:stop/10
+    a = collect(start:step:stop)./10
+    ra = collect(r)
+
+    @test r == a
+    @test isequal(r, a)
+
+    @test r == ra
+    @test isequal(r, ra)
+
+    @test hash(r) == hash(a)
+    @test hash(r) == hash(ra)
+
+    if len > 0
+        l = linspace(start/10, stop/10, len)
+        la = collect(l)
+
+        @test a == l
+        @test r == l
+        @test isequal(a, l)
+        @test isequal(r, l)
+
+        @test l == la
+        @test isequal(l, la)
+
+        @test hash(l) == hash(a)
+        @test hash(l) == hash(la)
+    end
+end
 
-@test [0.1:0.1:0.3;]   == [linspace(0.1,0.3,3);]     == [1:3;]./10
-@test [0.0:0.1:0.3;]   == [linspace(0.0,0.3,4);]     == [0:3;]./10
-@test [0.3:-0.1:-0.1;] == [linspace(0.3,-0.1,5);]    == [3:-1:-1;]./10
-@test [0.1:-0.1:-0.3;] == [linspace(0.1,-0.3,5);]    == [1:-1:-3;]./10
-@test [0.0:0.1:1.0;]   == [linspace(0.0,1.0,11);]    == [0:10;]./10
-@test [0.0:-0.1:1.0;]  == [linspace(0.0,1.0,0);]     == []
-@test [0.0:0.1:-1.0;]  == [linspace(0.0,-1.0,0);]    == []
-@test [0.0:-0.1:-1.0;] == [linspace(0.0,-1.0,11);]   == [0:-1:-10;]./10
```

Comment:
Good catch. I'm not sure what these are testing (I rewrote these tests last year), so I've added them to the loop just in case.